core/services/vrf/v2/integration_helpers_test.go

```diff
@@ -1720,3 +1742,42 @@ func testMaliciousConsumer(
 	}
 	require.Equal(t, 1, len(requests))
 }
+
+func getMetricInt64(c prometheus.Collector) (int64, error) {
+	var (
+		m      prometheus.Metric
+		mCount int
+		mChan  = make(chan prometheus.Metric)
+		done   = make(chan struct{})
+	)
+
+	go func() {
+		for m = range mChan {
+			mCount++
+		}
+		close(done)
+	}()
+
+	c.Collect(mChan)
+	close(mChan)
+	<-done
+
+	if mCount != 1 {
+		return 0.0, fmt.Errorf("collected %d metrics instead of exactly 1", mCount)
+	}
+
+	pb := &dto.Metric{}
+	if err := m.Write(pb); err != nil {
+		panic(fmt.Errorf("error happened while collecting metrics: %w", err))
```

Comment:
Why panic?

Reply:
I copy/pasted this code from the prometheus code, they had this panic in there. What's the recommended alternative?

Reply:
I guess we could just return the error and assert it's nil.

Reply:
Yeah if the signature already supports it, so as not to disrupt the whole test suite.